frontend/src/features/collections/edit-dialog/general-section.ts

```diff
@@ -0,0 +1,320 @@
+          }}
+        >
+          <div
+            class="flex size-full flex-col items-center justify-center bg-cover"
+            style="background-image:url('${path}')"
+          >
+            ${isSelected
+              ? html`<sl-icon
+                  class="size-10 stroke-black/50 text-white drop-shadow-md [paint-order:stroke]"
+                  name="check-lg"
+                ></sl-icon>`
+              : nothing}
+          </div>
+        </button>
+      </sl-tooltip>
+    `;
+  };
+
+  return html`
+    <fieldset
+      role="radiogroup"
+      aria-labelledby="collection-thumbnail-selector"
+      class="contents"
+    >
+      <label
+        id="collection-thumbnail-selector"
+        class="form-label flex items-center gap-1.5"
+      >
+        ${msg("Thumbnail")}
+        <sl-tooltip
+          content=${msg(
+            "Choose a thumbnail to represent this collection in the org dashboard and profile page.",
+          )}
+        >
+          <sl-icon name="info-circle"></sl-icon>
+        </sl-tooltip>
+      </label>
+      <div class="-mx-4 -my-2 -mt-1 overflow-x-auto px-4 py-2">
+        <div class="grid w-max min-w-full grid-rows-[auto_auto] gap-3">
+          <div class="row-start-1 text-xs text-neutral-500">
+            ${msg("Page Thumbnail")}
+          </div>
+          ${renderPageThumbnail.bind(this)(
+            this.defaultThumbnailName == null
+              ? this.collection?.thumbnail?.path
+              : null,
+          )}
+          <div class="sticky left-0 row-start-1 text-xs text-neutral-600">
+            ${msg("Placeholder")}
+          </div>
+          ${thumbnail(Thumbnail.Cyan)} ${thumbnail(Thumbnail.Green)}
+          ${thumbnail(Thumbnail.Yellow)} ${thumbnail(Thumbnail.Orange)}
+        </div>
+      </div>
+    </fieldset>
+  `;
+}
+
+function renderPageThumbnail(
+  this: CollectionEdit,
+  initialPath?: string | null,
+) {
+  const replaySource = `/api/orgs/${this.orgId}/collections/${this.collection!.id}/replay.json`;
+  // TODO Get query from replay-web-page embed
+  const query = queryString.stringify({
+    source: replaySource,
+    customColl: this.collection!.id,
+    embed: "default",
+    noCache: 1,
+    noSandbox: 1,
+  });
+
+  const isSelected = this.defaultThumbnailName == null;
+
+  this.thumbnailPreview?.thumbnailBlob
+    .then((value) => {
+      this.blobIsLoaded = !!value;
+    })
+    .catch(() => {
+      this.blobIsLoaded = false;
+    });
+
+  const enabled =
+    (!!this.selectedSnapshot && this.blobIsLoaded) || !!initialPath;
+
+  return html`
+    <button
+      class=${clsx(
+        isSelected ? tw`ring-2 ring-blue-300` : tw` ring-1 ring-stone-600/10`,
+        tw`row-start-2 aspect-video min-w-48 overflow-hidden rounded transition-all hover:ring-2 hover:ring-blue-300`,
+      )}
+      role="radio"
+      type="button"
+      aria-checked=${isSelected}
+      @click=${() => {
+        this.thumbnailSelector?.input?.focus();
```

Comment:
Seeing an issue where if I click and then "Page URL", or vice versa, the dropdown for "Page URL" doesn't open.

Reply:
Can reproduce, looking into this